### DISCUSSION TITLE
Should we use the custom withdrawal transaction fee value for DAO governance transaction mining fees?

### DISCUSSION THREAD

**Discussion by cbeams** [bisq-network/bisq#5269] [Q&A] [answered]
Currently, DAO governance transactions like submitting compensation requests and voting require the user to pay the mining fee rate reported by pricenodes.

With high fee rates, this poses a big disincentive to voting in particular.

I think we should prefer using the custom withdrawal fee rate (if it has been set). This would allow users to exercise their own judgement about how much they're willing to spend to ensure that their proposals and votes get in on time.

If we don't do this, then it's very likely that people will just stop voting because it's uneconomical to do so. That may have already started happening, I don't know.

For a concrete example, as I write this, I am trying to submit [my compensation request for Cycle 22](https://github.com/bisq-network/compensation/issues/794). I am requesting only 42 BSQ there, equivalent to 50 USD. The current fee rate reported by my pricenode is 70 sat/byte, meaning that I would pay ~10 USD to submit a 50 USD proposal. That doesn't make any sense, so I'm now forced to periodically check my Bisq client to see when the fee rate drops to an acceptable level for me (which would be around 30 sat/byte). There is a real risk that I will just forget about this, and that my compensation request won't get submitted at all. And in any case, it represents annoying busy work for me that's not really worth the 50 bucks I'm trying to get in the first place. The situation will be even worse when it comes to voting, because that's pure cost; I won't directly "get anything back" for submitting my (probably expensive) vote.

I would much rather be able to just set my custom withdrawal fee at 30 sat/byte and submit my compensation request using that value. I have a high degree of confidence based on recent mempool dynamics that this value will be high enough to get my compensation request into a block before the submission phase deadline. Why not let contributors and stakeholders make that choice for themselves?

**Comment by BtcContributor** [comment]
Great proposal.

Totally agree!

**Comment by MwithM** [comment]
As Bisq contributors must have knowledge about how blockspace market works, Bisq could let them choose it.
I'd like to add that I think that recurrent contributions like yours, under 100USD, might wait for another cycle to get the compensation to increase the amount. You should place the Github request but not submit it to the DAO and then bring it for the next Cycle.

**Comment by chimp1984** [comment, marked as answer]
Using too low minerfee comes with the risk that it will not get into the blockchain in the required phase and then you need to withdraw the BTC from your BSQ wallet back to your BTC wallet (the BTC for the requested BSQ). So you end up worse in that case.
I agree to @MwithM that best is to batch smaller amount requests.

The overall problem with high miner fees are for sure a serious problem for Bisq at every level. Unfortunately there is no easy solution for that. I think we have to accept that migrating off from the Bitcoin blockchain might be the only way forward. But that is big effort and requires more research to make the right decision which solutions would fit best for our needs.
The DAO could be swapped to a new asset by proveable burning BSQ on the BTC blockchain and by that getting issued new BSQ on the new chain. But also here it would require lot of research to pick the best candidate and will be lot of effort.

**Reply by cbeams** [reply]
> Using too low minerfee comes with the risk that it will not get into the blockchain in the required phase and then you need to withdraw the BTC from your BSQ wallet back to your BTC wallet (the BTC for the requested BSQ). So you end up worse in that case.

Good point. Batching may be a good enough solution for small compensation requests. Doesn't handle the voting case of course. I'll consider this discussion answered now in any case.
